### PR TITLE
Docker support for eZ Commerce

### PR DESCRIPTION
**Pull request by vidarl**
This PR fixes the docker and docker-compose configs

### CODE REVIEW

**Comment by vidarl** [comment]
Complete, but [this](https://github.com/ezsystems/ezplatform/pull/343) pr should be merged to this repo first

File: app/config/config.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ doctrine:
                 user: '%database_user%'
                 password: '%database_password%'
                 charset: utf8mb4
+                server_version: '%database_server_version%'
```
Comment (andrerom) [inline app/config/config.yml]: This changes should probably also go to ezplatform repo, + change in .platform/services.yaml + .env to make people aware they need to adjust config for this, if they adjust version there.

File: app/config/parameters.yml.dist [hunk 2]
```diff
@@ -16,6 +16,7 @@ parameters:
     env(DATABASE_NAME): ezcommerce
     env(DATABASE_USER): root
     env(DATABASE_PASSWORD):
+    env(DATABASE_SERVER_VERSION): mariadb-10.2.12
```
Comment (andrerom) [inline app/config/parameters.yml.dist]: Maybe a inline note above that if this is adjusted, so should platform.sh / docker config if used.

File: app/config/config.yml (hunk 1 above)
Comment (vidarl) [reply app/config/config.yml]: Sure about that? AFAIK, server_version is not required by ezplatform ATM

Comment (andrerom) [reply app/config/config.yml]: Not sure at all.

File: doc/docker/Dockerfile-distribution [hunk 3]
```diff
@@ -1,6 +1,7 @@
-# Note : if you set the environment variable COMPOSE_PROJECT_NAME to a non-default value, you'll need to change the
-# image name in here too
-FROM docker_app as builder
+# Note : if you set the environment variable COMPOSE_PROJECT_NAME to a non-default value, you'll need to set the
+# DISTRIBUTION_IMAGE build arg too (for instance docker-compose build --no-cache --build-arg DISTRIBUTION_IMAGE=customprojectname_app distribution)
+ARG DISTRIBUTION_IMAGE=docker_app
+FROM ${DISTRIBUTION_IMAGE} as builder
```
Comment (andrerom) [inline doc/docker/Dockerfile-distribution]: these changes are now merged up, rebase to cleanup

File: doc/docker/Dockerfile-solr [hunk 4]
```diff
@@ -1,19 +1,17 @@
...
+ && cp -R /opt/solr/server/migrations/solr/* /opt/solr/server/ez/template \
  && cp /opt/solr/server/solr/configsets/basic_configs/conf/currency.xml /opt/solr/server/ez/template \
  && cp /opt/solr/server/solr/configsets/basic_configs/conf/solrconfig.xml /opt/solr/server/ez/template \
  && cp /opt/solr/server/solr/configsets/basic_configs/conf/stopwords.txt /opt/solr/server/ez/template \
  && cp /opt/solr/server/solr/configsets/basic_configs/conf/synonyms.txt /opt/solr/server/ez/template \
  && cp /opt/solr/server/solr/configsets/basic_configs/conf/elevate.xml /opt/solr/server/ez/template \
  && cp /opt/solr/server/solr/solr.xml /opt/solr/server/ez \
  && sed -i.bak '/<updateRequestProcessorChain name="add-unknown-fields-to-the-schema">/,/<\/updateRequestProcessorChain>/d' /opt/solr/server/ez/template/solrconfig.xml \
- && sed -ie 's/${solr.autoSoftCommit.maxTime:-1}/${solr.autoSoftCommit.maxTime:20}/' /opt/solr/server/ez/template/solrconfig.xml
+ && sed -i.bak2 's/${solr.autoSoftCommit.maxTime:-1}/${solr.autoSoftCommit.maxTime:20}/' /opt/solr/server/ez/template/solrconfig.xml
```
Comment (andrerom) [inline doc/docker/Dockerfile-solr]: should these  changes _(all on  doc/docker/Dockerfile-solr)_ be on upstream?

Comment (vidarl) [reply doc/docker/Dockerfile-solr]: done

Comment (vidarl) [reply doc/docker/Dockerfile-solr]: FYI : Added related PR : https://github.com/ezsystems/ezcommerce/pull/6

File: .env [hunk 5]
```diff
@@ -22,7 +22,7 @@ APP_DOCKER_FILE=doc/docker/Dockerfile-app
 DATASET_VARDIR=my-ez-app
 
 # Install config, used by .platform.app.yaml among others
-INSTALL_EZ_INSTALL_TYPE=ezplatform-ee-clean
+INSTALL_EZ_INSTALL_TYPE=ezcommerce_installer
```
Comment (andrerom) [inline .env]: Side: Guess we can get rid of this env by taking advantage of the composer alias we have on all repos now.

**Comment by andrerom** [comment]
@vidarl Feel free to merge and merge up to master (and advance repo once you need it there)

File: app/config/parameters.yml.dist (hunk 2 above)
Comment (vidarl) [reply app/config/parameters.yml.dist]: ref. https://doc.ezplatform.com/en/2.3/getting_started/requirements/ we should maybe use "mysql-5.5.3" as default value ?

Comment (andrerom) [reply app/config/parameters.yml.dist]: hmm, that would breka with what we set in docker and p.sh config, kind of need to be the same as there I think.

Or we need to set it to null (if possible), and rather autodetect in cases where we can _(p.sh on v2 infra afaik, and on docker if we add env info about it or detect it in php image startup script or something)_

Comment (vidarl) [reply app/config/parameters.yml.dist]: I believe DATABASE_SERVER_VERSION just tells the ORM what kind of feature sets are available in db. So having it set to a low version, might impact performance ( or maybe fail if some doctrine ORM features are not supported by elder db versions).

However, there are little documentation about this

**Pull request merged**